src/java/org/apache/cassandra/cql3/constraints/ColumnConstraints.java

```diff
@@ -0,0 +1,201 @@
+{
+    public static final Serializer serializer = new Serializer();
+    public static final Noop NO_OP = new Noop();
+
+    private static final Set<Class<? extends AbstractType>> UNSUPPORTED_TYPES = ImmutableSet.of(MapType.class,
+                                                                                                TupleType.class,
+                                                                                                UserType.class,
+                                                                                                CompositeType.class,
+                                                                                                DynamicCompositeType.class);
+
+
+
+    private final List<ColumnConstraint<?>> constraints;
+    private final int constraintsSize;
+    private final boolean isEmpty;
+
+    public ColumnConstraints(List<ColumnConstraint<?>> constraints)
+    {
+        this.constraints = new ArrayList<>(constraints);
+        // These are catched values to avoid making the calculations every time
+        this.constraintsSize = constraints.size();
+        this.isEmpty = constraints.isEmpty();
+    }
+
+    @Override
+    public IVersionedSerializer<ColumnConstraints> serializer()
+    {
+        return serializer;
+    }
+
+    @Override
+    public void appendCqlTo(CqlBuilder builder)
+    {
+        for (ColumnConstraint<?> constraint : constraints)
+            constraint.appendCqlTo(builder);
+    }
+
+    @Override
+    public void evaluate(Class<? extends AbstractType> valueType, Object columnValue) throws ConstraintViolationException
+    {
+        for (ColumnConstraint<?> constraint : constraints)
+            constraint.evaluate(valueType, columnValue);
+    }
+
+    public List<ColumnConstraint<?>> getConstraints()
+    {
+        return constraints;
+    }
+
+    public boolean isEmpty()
+    {
+        return isEmpty;
+    }
+
+    public int getSize()
+    {
+        return constraintsSize;
+    }
+
+    @Override
+    public void validate(ColumnMetadata columnMetadata) throws InvalidConstraintDefinitionException
+    {
+        if (UNSUPPORTED_TYPES.contains(columnMetadata.type.getClass()))
+            throw new InvalidConstraintDefinitionException("Constraint cannot be defined on column '"
+                                                           + columnMetadata.name + "' with type: "
+                                                           + columnMetadata.type.asCQL3Type());
+
+        for (ColumnConstraint<?> constraint : constraints)
+            constraint.validate(columnMetadata);
+    }
+
+    @Override
+    public ConstraintType getConstraintType()
+    {
+        return ConstraintType.COMPOSED;
+    }
+
+    public static class Noop extends ColumnConstraints
+    {
+        private Noop()
+        {
+            super(Collections.emptyList());
+        }
+
+        @Override
+        public boolean isEmpty()
```

Comment:
@bbotella by the way ... since we are caching that in constructor, you dont need to override this anymore. You would need to do that only in case you would e.g. passed null list into the constructor, but that is not possible in our case.

Reply:
@bbotella I am actually questioning the existence of this Noop class as such. Because you could do just perfectly fine 

````
public static final ColumnConstraints NO_OP = new ColumnConstraints(Collections.emptyList());
````

and be done with it.

Reply:
@bbotella 

On the other hand, taking more deep look at that, what Noop has to do is to override `validate`, because for Noop constraint, every ColumnMetadata would be valid.

So my advice here is to keep `Noop` class, make it private, change `Noop` type of `NO_OP` to `ColumnConstraints`, remove `isEmpty()` and `getSize()` and override `validate` - the method would do literally nothing, it would be just overriden.